.github/workflows/matrix.yml

```diff
@@ -11,7 +11,7 @@ on:
 # We only care about the latest revision of a PR, so cancel all previous instances.
 concurrency:
   group: general-build-${{ github.event.pull_request.number || github.ref_name }}
-  cancel-in-progress: true
+  cancel-in-progress: ${{ github.event_name == 'pull_request' }}
```

Comment:
My turn to be (un-)professionally embarrassed for not realizing the solution was as easy as this.

Reply:
I was stumped for about 10 minutes, you're not alone.